ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/Runner.java

```diff
@@ -60,53 +61,55 @@
   private final HttpResponseFactory responseFactory = new HttpResponseFactory();
   private final Path dataDirectory;
 
-  private final Vertx vertx;
+  private Vertx vertx;
   private String jsonRpcHttpServiceId;
   private HttpServerService httpServerService;
 
   public Runner(
       final TransactionSerialiser serialiser,
-      final Vertx vertx,
       final HttpClientOptions clientOptions,
       final HttpServerOptions serverOptions,
       final Duration httpRequestTimeout,
       final TransactionFactory transactionFactory,
       final Path dataDirectory) {
     this.serialiser = serialiser;
     this.clientOptions = clientOptions;
+    LOG.debug("CLIENT OPTIONS: clientOptions.toString(): " + clientOptions.toString());
     this.serverOptions = serverOptions;
     this.httpRequestTimeout = httpRequestTimeout;
     this.transactionFactory = transactionFactory;
     this.dataDirectory = dataDirectory;
-    this.vertx = vertx;
   }
 
   public void start() {
+    this.vertx = Vertx.vertx();
     httpServerService = new HttpServerService(router(), serverOptions);
     vertx.deployVerticle(httpServerService, this::httpServerServiceDeployment);
   }
 
   public void stop() {
-    vertx.undeploy(jsonRpcHttpServiceId);
+    if (vertx != null) {
+      vertx.close();
+    }
   }
 
-  private RequestMapper createRequestMapper(final Vertx vertx) {
+  private RequestMapper createRequestMapper() {
 
     final HttpClient downStreamConnection = vertx.createHttpClient(clientOptions);
 
     final RequestMapper requestMapper =
         new RequestMapper(
             new PassThroughHandler(
                 downStreamConnection,
-                responseBodyHandler ->
+                (VertxRequestTransmitter.ResponseBodyHandler responseBodyHandler) ->
```

Comment:
Is this needed? (I haven't seen this done before)

Reply:
Yes ... I wasn't sure about it. IntelliJ gave me warnings about this, so I "fixed" it. IntelliJ likes to see the parameter type expanded.

Reply:
Done